### DISCUSSION TITLE
Endpoints of realtime?

### DISCUSSION THREAD

**Discussion by denchic45** [supabase/supabase#11112] [Questions] [answered]
I need realtime functionality, but I did not find an open api anywhere except for client libraries. I'm going to listen to data through a proxy server.

**Comment by GaryAustin1** [comment]
Realtime uses websockets.  You would probably need to pick a client library and use that source code as a reference.

**Reply by denchic45** [reply]
But after all theoretically it is possible to use websockets without client libraries? With my proxy server, using the client library will become a problem

**Reply by denchic45** [reply]
it would be nice if websocket endpoints were documented

**Reply by GaryAustin1** [reply]
Yes, I'm just saying you would need to look at the source of a client library to see what you need to do, not use the client library.
I've not seen docs on the websocket interface, set up, etc, but it does use Phoenix.

**Comment by w3b6x9** [comment, marked as answer]
@denchic45 we have prioritized getting the Realtime OpenAPI out and that'll be done in the near future. We had a section in Realtime v1 README where we specify what the WebSocket URL looks like: https://github.com/supabase/realtime/tree/v1#websocket-connection-authorization. I'll add that back for v2 README on `main`. You can also refer to this issue as it provides some insight: https://github.com/supabase/realtime/issues/457.
